tests/test_models/test_fields.py

```diff
@@ -45,10 +50,50 @@ def test_save_method(existing_artist: Artist, mocker: MockerFixture):
     assert existing_artist.bands.count() == 2
 
 
-def test_save_readonly_field(existing_artist: Artist):
+def test_save_readonly_field(existing_artist: Artist, band: Band):
     """Simple test to check that readonly field changes nothing."""
+    factories.MembershipFactory(artist=existing_artist, band=band)
     field = IntermediateManyToManyField(attribute="bands", readonly=True)
     field.save(existing_artist, {})
+    assert existing_artist.bands.all()
+
+
+def test_save_with_exception_during_full_clean(
+    existing_artist: models.Model,
```

Comment:
And below for other tests
```suggestion
    existing_artist: Artist,
```